types/lodash/index.d.ts

```diff
@@ -12059,7 +12063,22 @@ declare namespace _ {
          * _.isArrayLike(_.noop);
          * // => false
          */
-        isArrayLike<T>(value?: any): value is T[];
+        isArrayLike<T>(value: T & string & number): boolean; // should only match if T = any
```

Comment:
This feels a bit hacky, so I'm open to input on this one. This was the only way I could make my `let value: any` test pass. The issue is that without this line, typescript will use this overload: ` isArrayLike<any>(value: T | Function): value is any`. And if the function returns false, typescript says that `value` is now type `never`, which is wrong.